In [13]:
import sqlite3 as sq3
import pandas as pd
import numpy as np
import statsmodels.api as sm

In [4]:
conn = sq3.connect('bike_data.db')
df = pd.read_sql_query("SELECT * from yelp_data", conn)
df

name  rating                      address  \
0                   Betty’s East     4.5          1301 Queen Street E   
1               Lloyd's On Queen     5.0              1298 Queen St E   
2                Boardwalk Place     2.5  1675 Lake Shore Boulevard E   
3   Baldev Paan Cold Drink House     3.0        1399 Gerrard Street E   
4                 Two Headed Dog     2.5        1469 Gerrard Street E   
5               The Thirsty Loon     4.0        1489 Gerrard Street E   
6                      Lake Inez     4.5        1471 Gerrard Street E   
7                     Black Pony     4.5        1481 Gerrard Street E   
8                    Jenni Place     3.0        1338 Gerrard Street E   
9                     Beach Club     3.0  1681 Lake Shore Boulevard E   
10            Toronto Beach Club     3.0  1681 Lake Shore Boulevard E   
11                 JP Restaurant     3.5           270 Coxwell Avenue   
12                 The Duke Live     3.5          1225 Queen Street E   
13    Shamrock Bowl and Restobar     4.0           280 Coxwell Avenue   
14               Le Conciliabule     4.5        1300 Gerrard Street E   
15                  Queen's Head     2.5          1214 Queen Street E   

     latitude  longitude    distance  
0   43.663455 -79.327992  689.778721  
1   43.663801 -79.328356  707.598820  
2   43.663010 -79.310898  758.567399  
3   43.671710 -79.323520  772.512252  
4   43.672358 -79.320782  792.227554  
5   43.672394 -79.319906  792.332218  
6   43.672340 -79.320640  794.661955  
7   43.672410 -79.320400  805.518073  
8   43.671800 -79.324810  829.759140  
9   43.663207 -79.310013  839.312073  
10  43.663207 -79.310013  839.312073  
11  43.673030 -79.319550  868.716398  
12  43.663149 -79.330307  877.705944  
13  43.673290 -79.319790  896.914859  
14  43.671365 -79.327225  903.609078  
15  43.663210 -79.330855  918.527449

## Hypothesis Testing

**H0**: There is not a correlation between distance of bike stop to a bar and the rating of the bar.

**HA**: There is a correlation between distance of bike stop to a bar and the rating of the bar.


---

Build a regression model.

In [20]:
X = df['distance']
y = df['rating']

X = sm.add_constant(X) # adding a constant
lin_reg = sm.OLS(y,X)

model = lin_reg.fit()
print_model = model.summary()
print(print_model)

                            OLS Regression Results                            
Dep. Variable:                 rating   R-squared:                       0.079
Model:                            OLS   Adj. R-squared:                  0.013
Method:                 Least Squares   F-statistic:                     1.200
Date:                Sun, 30 Jul 2023   Prob (F-statistic):              0.292
Time:                        19:47:32   Log-Likelihood:                -18.758
No. Observations:                  16   AIC:                             41.52
Df Residuals:                      14   BIC:                             43.06
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          6.4839      2.646      2.450      0.0

C:\ProgramData\anaconda3\Lib\site-packages\scipy\stats\_stats_py.py:1736: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=16
  warnings.warn("kurtosistest only valid for n>=20 ... continuing "


## Provide model output and an interpretation of the results.

---

**R-Squared**: 0.936, this is saying that the model can explain 7.9% of the data, it seems like a very poor fit, this likely telling us that
there is little to no correlation between how far a bar is from a bike stop and the rating of that bar.

**coef**: the coef of distance is saying that for every 1 unit increase in distance, rating will decrease by -0.0035.

**P>|t|**: this value is 0.292 which means there is means there is plently of evidence that our null hypothesis is true therefore we can conclude that data turned out the way did soley due to natrual variance.

Note: we used a very small amount of data in this model and only from one bikestop, these results can still be explored further but it
is important to take note of the sample size.

## Buisness Context of results

The results of the data analysis safely concludes that business owners should not concern themselves with having bike stations installed near their bars if their goal is into increase ratings of their business.

# Stretch

How can you turn the regression model into a classification model?